app/code/Magento/Store/App/Request/PathInfoProcessor.php

```diff
@@ -38,7 +38,20 @@ public function process(\Magento\Framework\App\RequestInterface $request, $pathI
             /** @var \Magento\Store\Api\Data\StoreInterface $store */
             $store = $this->storeManager->getStore($storeCode);
         } catch (NoSuchEntityException $e) {
-            return $pathInfo;
+            try {
+                /** get the store by url */
+                $requestUri = $request->getUri();
+                if (empty($requestUri)) {
+                    return $pathInfo;
+                }
+                $store = $this->storeManager->getStoreByUrl($requestUri);
+                $storeBaseUrl = $store->getBaseUrl();
+                $baseUrlUri = \Zend\Uri\UriFactory::factory($storeBaseUrl);
+                $baseUrlPath = $baseUrlUri->getPath();
+                $pathInfo = preg_replace('#^' . $baseUrlPath . '#', '', $pathInfo);
```

Comment:
`preg_replace` is misused here. Just compare string beginning and then do `str_replace`.